crates/tools/src/lib.rs

```diff
@@ -116,3 +120,28 @@ fn install_rustfmt() -> Result<()> {
         ".",
     )
 }
+
+pub fn install_format_hook() -> Result<()> {
+    let path = Path::new("./.git/hooks/pre-commit");
+    if !path.exists() {
+        let mut open_options = OpenOptions::new();
+        #[cfg(unix)]
+        {
+            // Set as executable
+            open_options.mode(0o770);
+        }
+        let mut file = open_options.write(true).create(true).open(path)?;
+        write!(
+            file,
+            r#"#!/bin/sh
+
+cargo format
+for path in $( git diff --name-only --cached ); do
+    git update-index --add $path
+done"#
```

Comment:
Hm, let's make a hook itself a rust program, so that it doesn't depend on a particular shell and works on windows?

What I have in mind is to have a `src/bin/pre-commit.rs` program which basically runs [this function](https://github.com/rust-analyzer/rust-analyzer/blob/1e554d551f1510780377158fda8d86ff8c8266fe/crates/tools/src/lib.rs#L94). 

Then the `install-format-hook` would run `cargo build --release --package tools --bin pre-commit` and copy the binary from `target/release` to `./git/hooks`